### ISSUE TITLE
Consistent timestamp output formatting

### ISSUE THREAD

**Issue by philrz** [brimsec/zq#1002]
[A user on public Slack observed](https://brimsec.slack.com/archives/CTSMAK6G7/p1594740349054600):

> is there a way to change the output timestamp format? decimal times don't really mean much as im looking at a table, i'd like to see it as iso8601

Indeed, things are a bit inconsistent in this area as of `zq` commit `85c31a3`. Here's the list of available output formats:

```
$ zq help 2>&1 | grep "format for output"
    -f format for output data [zng,ndjson,table,text,types,zeek,zjson,tzng] (default "zng")
```

Of these, `ndjson`, `table`, `text`, and the just-added `csv` all seem like formats where the user might want either format.

We see that `ndjson` and `table` currently default to epoch.

```
$ zq -f ndjson 'cut ts | head 1' stats.log.gz 
{"ts":"1521911720.600725"}

$ zq -f table 'cut ts | head 1' stats.log.gz 
TS
1521911720.600725
```

Whereas `text` and `csv` default to ISO8601.

```
$ zq -f text 'cut ts | head 1' stats.log.gz 
2018-03-24T17:15:20.600725Z

$ zq -f csv 'cut ts | head 1' stats.log.gz 
ts
2018-03-24T17:15:20.600725Z
```

However, there's also an option to change these to epoch.

```
$ zq help 2>&1 | grep "epoch"
    -E display epoch timestamps in csv and text output (default "false")

$ zq -E -f text 'cut ts | head 1' stats.log.gz 
1521911720.600725

$ zq -E -f csv 'cut ts | head 1' stats.log.gz 
ts
1521911720.600725
```

The proposal is to standardize on this latter behavior such that `ndjson` and `table` also output ISO8601 by default, but their current behavior could also be restored by the user via `-E`.

(Side note: The Brim app has its "Preferences" menu that lets the user set this formatting, but it's not currently possible in `zq`. In the past I've argued that issues like this are a reminder that we could at some point implement persistent preferences in `zq`. We're slowly accumulating command-line options that I could imagine users thinking "Yeah, I want it set like that all the time." But I respect that's out-of-scope of this specific issue.)

**Comment by captainGeech42** [comment]
+1 for a config file or something for persistent options, that'd be great

**Comment by mikesbrown** [comment]
Note as a workaround, `zq -f text` will give you ISO 8601.

**Comment by philrz** [comment]
In a group discussion today it was noted that the NDJSON output code is not wired up currently in a way that would make it easy to offer multiple output format options there. Since we have active user requests for ISO8601 format, and many other systems that output JSON tend to format their timestamps as ISO8601, the current plan is to switch the NDJSON output to that. If users then come out of the woodwork saying they were big fans of the epoch float format, we'd then put in the effort to bring the flexibility to choose multiple formats.

**Comment by philrz** [comment]
Verified in `zq` commit `ba75962e`.

Revisiting the formats discussed above, they all now output ISO timestamps by default (`text` & `csv` already did, and `ndjson` and `table` now do thanks to the changes in the linked PRs).

```
$ zq -f ndjson 'cut ts | head 1' stats.log.gz
{"ts":"2018-03-24T17:15:20.600725Z"}

$ zq -f table 'cut ts | head 1' stats.log.gz
TS
2018-03-24T17:15:20.600725Z

$ zq -f text 'cut ts | head 1' stats.log.gz
2018-03-24T17:15:20.600725Z

$ zq -f csv 'cut ts | head 1' stats.log.gz
ts
2018-03-24T17:15:20.600725Z
```

If the epoch float is ever preferred, the `-E` option is available for three of these.

```
$ zq help 2>&1 | grep "epoch"
    -E display epoch timestamps in csv, table, and text output (default "false")

$ zq -E -f text 'cut ts | head 1' stats.log.gz
1521911720.600725

$ zq -E -f csv 'cut ts | head 1' stats.log.gz
ts
1521911720.600725

$ zq -E -f table 'cut ts | head 1' stats.log.gz
TS
1521911720.600725
```

As disclosed in a comment above, the `-E` option is not currently available for the `ndjson` output format. This hopefully is ok since so many JSON outputs from other tools print ISO timestamps. If users speak up and say they really need epoch back for it, we'll reconsider.

Thanks @nwt!

**Comment by philrz** [comment]
Here's additional verification in `zq` commit `df30be8d` of what was additionally addressed in #1557.

As of `zq` commit `1e4f1553` that was right before #1557 was merged, fields of ZNG type `time` with names other than `ts` were still being output as epoch seconds.

```
$ zq -version
Version: v0.22.0-92-g1e4f1553

$ zq -f text 'cut certificate.not_valid_after' x509.log.gz
1678565702
1834731378
1678565702
...
```

Now in `zq` commit `df30be8d`, it's output as an ISO timestamp by default, with the `-E` option available to bring back epoch, if desired.

```
$ zq -version
Version: v0.22.0-93-gdf30be8d

$ zq -f text 'cut certificate.not_valid_after' x509.log.gz
2023-03-11T20:15:02Z
2028-02-21T07:36:18Z
2023-03-11T20:15:02Z
...

$ zq -E -f text 'cut certificate.not_valid_after' x509.log.gz
1678565702.000000
1834731378.000000
1678565702.000000
...
```

Thanks again @nwt!
